pkg/services/store/entity/db/dbimpl/dbimpl.go

```diff
@@ -54,26 +51,8 @@ func (db *EntityDB) GetEngine() (*xorm.Engine, error) {
 
 	// if explicit connection settings are provided, use them
 	if dbType != "" {
-		dbHost := cfgSection.Key("db_host").MustString("")
-		dbName := cfgSection.Key("db_name").MustString("")
-		dbUser := cfgSection.Key("db_user").MustString("")
-		dbPass := cfgSection.Key("db_pass").MustString("")
-
 		if dbType == "postgres" {
-			// TODO: support all postgres connection options
-			dbSslMode := cfgSection.Key("db_sslmode").MustString("disable")
-
-			addr, err := util.SplitHostPortDefault(dbHost, "127.0.0.1", "5432")
-			if err != nil {
-				return nil, fmt.Errorf("invalid host specifier '%s': %w", dbHost, err)
-			}
-
-			connectionString := fmt.Sprintf(
-				"user='%s' password='%s' host='%s' port='%s' dbname='%s' sslmode='%s'", // sslcert='%s' sslkey='%s' sslrootcert='%s'",
```

Comment:
The issue was the single quotes here. When we [parse the db name from the connection string](https://github.com/grafana/grafana/blob/ef51a64b5753bdd9e3e35703ff9899512f81c698/pkg/services/sqlstore/migrator/postgres_dialect.go#L343) we don't expect there to be quotes around the values.

Reply:
ugh, that's nasty. It may be a good idea to update that regex to allow quotes, or use something that can parse it properly, but this workaround is fine for now imho